### ISSUE TITLE
Clean up template code (mainly viewtopic)

### ISSUE THREAD

**Issue by javiexin** [Wolfsblvt/advancedpolls#56]
Simplify the template modifications by utilizing new events in viewtopic poll block (to be proposed):
Remove the use of "overall_header" event when more specific events may be used.
Remove the need for some JS code by asking for new events or template modifications.
Simplify the PHP code to remove generating HTML "hacks" and moving that to (new) template events.

**Comment by javiexin** [comment]
Already working on this refactoring exercise.

**Comment by Wolfsblvt** [comment]
I will create a new branch for that stuff later, after we have done most of the things for the next version. I will use that as a base.

**Comment by javiexin** [comment]
I have this code ready.  It requires some changes to viewtopic_body.html, but clarifies and simplifies our code.  We should chat to see if we propose the changes.
How do I share with you the modified viewtopic_body.html?

**Comment by Wolfsblvt** [comment]
Hmh, good question. You can put it on https://gist.github.com/, and I can check the changes in one of my diff tools.

**Comment by javiexin** [comment]
https://gist.github.com/javiexin/b0ca1c208037fcc43231

**Comment by Wolfsblvt** [comment]
So I should just check if I agree with your changes there?
I will note it and do it when I have time.

**Comment by javiexin** [comment]
Yes, I will explain what I use, and what I have included "just" for completeness.

**Comment by javiexin** [comment]
All changes are on the left margin (no indenting, on purpose).  Of those, I am only using some, namely:
viewtopic_body_poll_title_before.html // somewhere outside any other markup in the poll
viewtopic_body_poll_length_after.html // to create messages, rather than "appending" everything to a single string/lang variable
viewtopic_body_poll_option_after.html // the main loop of poll options
viewtopic_body_poll_submit_before.html // for poll totals without changing other labels

And the one thing that is more important for you to check is the "", that is critical, as what I am proposing is a new mechanism to interact with templates.  See how I replace the full block by using that.

**Comment by javiexin** [comment]
Forget this gist, I will do another one, showing the diffs in revisions.

**Comment by javiexin** [comment]
Use this one: https://gist.github.com/javiexin/45d6efabdcb205a70e4b

Three commits, first is the original phpbb 3.1.4 code, second is my proposed changes, third is the way I have used these changes in our extension, to show the possibilities and the need for this.

**Comment by Wolfsblvt** [comment]
Thanks for reporting this and for using the project. This project is no longer maintained, so I’m closing all remaining issues as part of archiving the repository. This avoids leaving a false impression that further development or support is planned.

Anyone interested in continuing development is welcome to do so through a fork, subject to the repository’s license and any applicable third-party licenses.

**Comment by Wolfsblvt** [comment]
Thanks for reporting this and for using the project. This repository is no longer maintained and is being archived, so I’m closing all remaining issues rather than leaving them open with the implication of future development or support. See the [Disclaimer](https://github.com/Wolfsblvt/advancedpolls#disclaimer) for details.
